xds/server.go

```diff
@@ -279,7 +407,16 @@ type listenerWrapper struct {
 	net.Listener
 	cancelWatch func()
 
-	// TODO(easwars): Add fields for certificate providers.
+	// A small race exists in the xdsClient code where an xDS response is
+	// received while the user is calling cancel(). In this small window the
+	// registered callback can be called after the watcher is canceled.
```

Comment:
Why is this comment relevant to this field?

Reply:
I've clarified it a bit.
Basically the xdsClient code documents this race and says that callers need to handle it.